test/Microsoft.Extensions.Caching.SqlServer.Tests/SqlServerCacheServicesExtensionsTest.cs

```diff
@@ -26,5 +29,87 @@ public void AddSqlServerCache_AddsAsSingleRegistrationService()
             Assert.Equal(typeof(SqlServerCache), serviceDescriptor.ImplementationType);
             Assert.Equal(ServiceLifetime.Singleton, serviceDescriptor.Lifetime);
         }
+
+        [Fact]
+        public void AddSqlServerCache_ReplacesPreviouslyUserRegisteredServices()
+        {
+            // Arrange
+            var services = new ServiceCollection();
+            services.AddScoped<IDistributedCache, TestDistributedCache>();
+
+            // Act
+            services.AddSqlServerCache(options => {
+                options.ConnectionString = "Fake";
+                options.SchemaName = "Fake";
+                options.TableName = "Fake";
+            });
+
+            // Assert
+            var serviceProvider = services.BuildServiceProvider();
+
+            var distributedCache = services.FirstOrDefault(desc => desc.ServiceType == typeof(IDistributedCache));
+
+            Assert.NotNull(distributedCache);
+            Assert.Equal(ServiceLifetime.Scoped, distributedCache.Lifetime);
+            Assert.IsType<SqlServerCache>(serviceProvider.GetRequiredService<IDistributedCache>());
+        }
+
+        private class TestDistributedCache : IDistributedCache
```

Comment:
I don't think so, we are basically resolving this services on the test. I don't know for sure if you could register a mock as an implementation for an interface on DI but It will make the test way more obscure.